operator/controllers/kyma_controller.go

```diff
@@ -419,6 +425,8 @@ func (r *KymaReconciler) SetupWithManager(mgr ctrl.Manager, options controller.O
 		}
 	}
 
+	controllerBuilder = listener.RegisterListenerComponent(logger, mgr, controllerBuilder, listenerAddr, &handler.EnqueueRequestForObject{}, componentName)
```

Comment:
1. controllerBuilder is passed as a pointer, so the value should not be overridden 
2. it doesn't make sense to pass the manager around. Rather you should get the channel from your listener you want to watch on
3. You listener should have its own logger instance
4. Instead of  `handler.EnqueueRequestForObject{} `use `handler.Funcs{GenericFunc: yourGenericFunc}`
Channel internally calls this, please check kubebuilder's implementation. 

`yourGenericFunc` will receive an eventHandler and a queue to add the kyma object too